### ISSUE TITLE
Required Words is Case Sensitive?

### ISSUE THREAD

**Issue by Flippage** [nosmokingbandit/watcher#166]
#### What happened?
I search for movies with "bluray, blu-ray, Bluray" in Required Words and no results are found. If I search with "BluRay" it finds 5 releases.

#### What was supposed to happen?
It should ignore text case when searching and find any release that has "bluray" in the title. Shouldn't need to include every different capitalization of the word "Bluray".

#### How can I reproduce this?
Search for a movie that you know has bluray in the title, and just capitalize a letter that isn't a capital in the release title, and watcher won't find it. You can use the examples in the logs below for testing.

Logs for search with "BluRay" and with all other spellings instead:

`INFO 2017-01-17 13:08:07,029 core.sqldb.update: Updating tt0387564 to {"Quality":{"4K":["false",0,"10000","50000"],"1080P":["true",1,"2000","10000"],"720P":["true",2,"1500","9000"],"SD":["false",3,"200","1000"]},"Filters":{"requiredwords":"BluRay","preferredwords":"bluray,blu-ray,DTS,DD5.1,DTS-HD","ignoredwords":"german,dutch,french,truefrench,danish,swedish,spanish,italian,korean,dubbed,swesub,korsub,dksubs,vain,HC,CAM,HDCAM,R4,R5,R6,DVDRip,SCREENER,NL,HC,HDTS,SUBS,HDTC,subbed,dubbed,FRENCH,GERMAN,HDRip,HD-TS"}} in MOVIES.
INFO 2017-01-17 13:08:07,048 core.sqldb.update: Updating tt0387564 to Wanted in MOVIES.
INFO 2017-01-17 13:08:07,104 core.sqldb.get_search_results: Retreving Search Results for tt0387564.
INFO 2017-01-17 13:08:07,128 core.sqldb.get_user_movies: Retreving list of user's movies.
INFO 2017-01-17 13:08:07,144 core.sqldb.get_search_results: Retreving Search Results for tt0387564.
INFO 2017-01-17 13:08:07,155 core.sqldb.get_user_movies: Retreving list of user's movies.
INFO 2017-01-17 13:08:08,644 core.sqldb.get_movie_details: Retreving details for tt0387564.
INFO 2017-01-17 13:08:08,645 core.sqldb.get_movie_details: Retreving details for tt0387564.
INFO 2017-01-17 13:08:08,647 core.newznab.search_all: SEARCHING: https://api.dognzb.cr/api?apikey=APIKEY&t=movie&imdbid=0387564
INFO 2017-01-17 13:08:13,763 core.newznab.search_all: SEARCHING: https://api.nzbgeek.info/api?apikey=APIKEY&t=movie&imdbid=0387564
INFO 2017-01-17 13:08:17,533 core.sqldb.get_search_results: Retreving Search Results for tt0387564.
INFO 2017-01-17 13:08:17,536 core.sqldb.get_movie_details: Retreving details for tt0387564.
INFO 2017-01-17 13:08:17,542 core.sqldb.get_marked_results: Retreving Marked Results for tt0387564.
INFO 2017-01-17 13:08:17,543 core.searcher.store_results: 5 results found for tt0387564. Storing results.
INFO 2017-01-17 13:08:17,562 core.sqldb.write_search_results: Writing batch into SEARCHRESULTS
INFO 2017-01-17 13:08:17,578 core.sqldb.get_distinct: Getting distinct values for tt0387564 in SEARCHRESULTS
INFO 2017-01-17 13:08:17,578 core.updatestatus.movie_status: Setting MOVIES tt0387564 status to Found.
INFO 2017-01-17 13:08:17,578 core.sqldb.update: Updating tt0387564 to Found in MOVIES.
INFO 2017-01-17 13:08:17,638 core.sqldb.get_search_results: Retreving Search Results for tt0387564.
INFO 2017-01-17 13:08:17,654 core.sqldb.get_user_movies: Retreving list of user's movies.
INFO 2017-01-17 13:15:02,078 core.sqldb.get_user_movies: Retreving list of user's movies.
INFO 2017-01-17 13:15:04,746 core.sqldb.get_movie_details: Retreving details for tt0387564.
INFO 2017-01-17 13:15:04,750 core.sqldb.get_search_results: Retreving Search Results for tt0387564.
INFO 2017-01-17 13:15:17,921 core.sqldb.get_movie_details: Retreving details for tt0387564.
INFO 2017-01-17 13:15:17,924 core.ajax.update_quality_settings: Updating quality for tt0387564.
INFO 2017-01-17 13:15:17,924 core.sqldb.update: Updating tt0387564 to {"Quality":{"4K":["false",0,"10000","50000"],"1080P":["true",1,"2000","10000"],"720P":["true",2,"1500","9000"],"SD":["false",3,"200","1000"]},"Filters":{"requiredwords":"bluray,Bluray,blu-ray","preferredwords":"bluray,blu-ray,DTS,DD5.1,DTS-HD","ignoredwords":"german,dutch,french,truefrench,danish,swedish,spanish,italian,korean,dubbed,swesub,korsub,dksubs,vain,HC,CAM,HDCAM,R4,R5,R6,DVDRip,SCREENER,NL,HC,HDTS,SUBS,HDTC,subbed,dubbed,FRENCH,GERMAN,HDRip,HD-TS"}} in MOVIES.
INFO 2017-01-17 13:15:17,941 core.sqldb.update: Updating tt0387564 to Wanted in MOVIES.
INFO 2017-01-17 13:15:17,983 core.sqldb.get_search_results: Retreving Search Results for tt0387564.
INFO 2017-01-17 13:15:18,016 core.sqldb.get_user_movies: Retreving list of user's movies.
INFO 2017-01-17 13:15:18,036 core.sqldb.get_search_results: Retreving Search Results for tt0387564.
INFO 2017-01-17 13:15:18,052 core.sqldb.get_user_movies: Retreving list of user's movies.
INFO 2017-01-17 13:15:19,976 core.sqldb.get_movie_details: Retreving details for tt0387564.
INFO 2017-01-17 13:15:19,977 core.sqldb.get_movie_details: Retreving details for tt0387564.
INFO 2017-01-17 13:15:19,979 core.newznab.search_all: SEARCHING: https://api.dognzb.cr/api?apikey=APIKEY&t=movie&imdbid=0387564
INFO 2017-01-17 13:15:31,188 core.newznab.search_all: SEARCHING: https://api.nzbgeek.info/api?apikey=APIKEY&t=movie&imdbid=0387564
INFO 2017-01-17 13:15:34,825 core.sqldb.get_search_results: Retreving Search Results for tt0387564.
INFO 2017-01-17 13:15:34,826 core.sqldb.get_movie_details: Retreving details for tt0387564.
INFO 2017-01-17 13:15:34,829 core.sqldb.get_marked_results: Retreving Marked Results for tt0387564.
INFO 2017-01-17 13:15:34,831 core.searcher.store_results: 0 results found for tt0387564. Storing results.`

**Comment by nosmokingbandit** [comment]
I'm not able to reproduce this. Using tt0387564 and no required words I got a few results. I added `bluray` to required, then got a result with `BluRay` in it.

So following your logs, you tried `BluRay` first, which worked for me and you:

`INFO 2017-01-17 13:08:17,543 core.searcher.store_results: 5 results found for tt0387564`

Then you tried `bluray,Bluray,blu-ray`, which got you zero results as intended. You are asking for the title of each movie to contain both `bluray` *and* `blu-ray`, which is probably not going to happen. Required words filtering is very strict.

What I'd suggest you do is require the word `blu`.

But, to answer your question more accurately, it is not case-sensitive:

scoreresults.py:50, 129
```
required = filters['requiredwords'].lower().split(u',')
[...]
results = [r for r in self.results if word in r['title'].lower()]
```

**Comment by Flippage** [comment]
Ahh, awesome. Yeah, I was expecting it to just need to have at least 1 of the required words. So that was just my understanding of how it worked. I was looking at what I had done, and I actually came to the same conclusion, that it was actually trying to find ALL of the required words which is why it was failing.

Either way, thank's for the fix. I think matching at least 1 of the required words is exactly how it should work :)

Great job dude.

Loving Watcher now that I've got it working haha.
